sdcm/cluster.py

```diff
@@ -2325,6 +2325,17 @@ def create_swap_file(self, size=1024):
         if "sct_configured_swapfile" not in result.stdout:
             self.log.warning("Swap file is not used on loader node %s.\nError details: %s", self, result.stderr)
 
+    def change_hostname(self):
+        self.log.debug('Changing hostname to %s', self.name)
+        result = self.remoter.run('sudo hostnamectl set-hostname {}'.format(self.name), ignore_status=True)
+        if result.ok:
+            self.log.info('Hostname has been changed. Rebooting...')
+            self.hard_reboot()
```

Comment:
do we really have to restart it ?

Reply:
wouldn't be enough to re-login/reconnect ?

Reply:
we are adding this for cosmetics reason, I'm not sure adding a reboot per node for this worth the time it would take.

Reply:
in manual, it is written to reboot, so all running processes get new hostname.
Will investigate more.

Reply:
one more idea it to try it in the startup_script, before we initially connect to the node.

Reply:
@fruch do we have something similar on gce?

Reply:
no, in gce it's automatically set the hostname by the instance name. don't know exactly how that happens.(google didn't told me :smile:)

Reply:
@fruch , with start_script it is a bit difficult, because we create instance before we create ( and generate) Node name

Reply:
@fruch just reconnecting doesn't help, because the database log continue to use old hostname:
Dec 10 11:22:58 ip-172-30-0-111.ec2.internal

Reply:
Exactly why I've suggested using the run script, todo it before configuration of syslogd

Reply:
@fruch I think adding the node name to start_script will require to much refactor (from my point of view): Now all instances have same start_script, if put code which set the hostname into it, we will need start_script per instances. Also for that moment we create all instances per cluster type with one call ec2_service.create_instances, otherwise it will require creating one by one.

Reply:
maybe just a restart to syslogd would be enough ? rebooting an i3 mean scylla data is lost, and might raise all kind of errors in the process, we should be careful with those.

Reply:
Restarting systemd-journald switch to new hostname:
```
-- Logs begin at Wed 2019-12-11 06:48:56 GMT. --
Dec 11 06:49:12 ip-172-30-0-142.ec2.internal systemd[1]: Starting Scylla AMI Setup...
Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 sudo[1864]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/sed -i s/#MaxSessions \(.*\)$/MaxSessions 1000/ /etc/ssh/sshd_config
Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 sudo[1868]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/systemctl restart sshd
Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 sudo[1874]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/sed -i s/#MaxSessions \(.*\)$/MaxSessions 1000/ /etc/ssh/sshd_config
Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 sudo[1880]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/systemctl restart sshd
Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 scylla_ami_setup[1547]: Waiting for cloud-init to finish...
Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 scylla_ami_setup[1547]: [INFO] Started with user data set to:
```

Reply:
> Restarting systemd-journald switch to new hostname:
> 
> ```
> -- Logs begin at Wed 2019-12-11 06:48:56 GMT. --
> Dec 11 06:49:12 ip-172-30-0-142.ec2.internal systemd[1]: Starting Scylla AMI Setup...
> Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 sudo[1864]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/sed -i s/#MaxSessions \(.*\)$/MaxSessions 1000/ /etc/ssh/sshd_config
> Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 sudo[1868]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/systemctl restart sshd
> Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 sudo[1874]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/sed -i s/#MaxSessions \(.*\)$/MaxSessions 1000/ /etc/ssh/sshd_config
> Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 sudo[1880]:     root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/systemctl restart sshd
> Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 scylla_ami_setup[1547]: Waiting for cloud-init to finish...
> Dec 11 06:49:26 abykov-test-abykov-db-node-e8b4df74-1 scylla_ami_setup[1547]: [INFO] Started with user data set to:
> ```

In my manual test, journalctl switched to new hostname without restarting systemd-journald.

Reply:
I ran several times, and it not happend.